internal/promotion/runner/builtin/git_pusher.go

```diff
@@ -181,11 +186,27 @@ func (g *gitPushPusher) run(
 		return promotion.StepResult{Status: kargoapi.PromotionPhaseErrored},
 			fmt.Errorf("error getting last commit ID: %w", err)
 	}
+
+	// Using git provider to get commit url.
+	gpOpts := &gitprovider.Options{}
+	gitProvider, err := gitprovider.New(workTree.URL(), gpOpts)
+	if err != nil {
+		return promotion.StepResult{Status: kargoapi.PromotionPhaseErrored},
+			fmt.Errorf("error creating git provider service: %w", err)
+	}
```

Comment:
This may not be how we want to deal with this error.

Everything else this step does can potentially succeed even if we're unable to construct this URL.

I think provider not found is the only thing that can actually go wrong in the call to `gitprovider.New()` so you probably just want to ignore this error entirely, although it's probably best to leave a comment about why ignoring it is the correct thing to do as well as a safe thing to do.

You can omit the URL from the output if no provider is found.

Reply:
Makes sense, remove the error.

Reply:
Would it make sense to add some log message to say we couldn't construct the URL? Just thinking that if users are trying to make use of it and it's not appearing in the output, how would they figure out why its being omitted?

Reply:
Yes. Good thought. And it's got me thinking now that it may even be good to set the URL to `Unknown` in this case. Without explicitly seeing that it didn't resolve, there might never be a thought to check the logs, so this would call attention to it more explicitly.